### DISCUSSION TITLE
custom bean and iso8583

### DISCUSSION THREAD

**Discussion by samar-elsayed** [apache/camel-karavan#1078] [Q&A]
Hi,

when I try to create a custom bean, do I need to create a new jar to be able to use it ? and how can I trigger it inside a processor ?
and finally does Marshal / Unmarshal component support iso8583 ?

**Comment by davsclaus** [comment]
It depends you can have custom java code together with karavan, and run it without having to create a JAR. However if you have a lot of java code its better to build this in a standard Java project and as a JAR, which you can then use with karavan.

There are some examples here (camel-jbang based) but karavan can run these also. Some of these examples does not use YAML DSL so Karavan cannot visualize/design these routes.

https://github.com/apache/camel-kamelets-examples/tree/main/jbang

For example here is some custom bean code
https://github.com/apache/camel-kamelets-examples/tree/main/jbang/bean-builder

**Reply by samar-elsayed** [reply]
@davsclaus can you tell me for example how can I create the bean tokenObj mentioned in the [answer here](https://stackoverflow.com/questions/45009438/rest-service-call-with-camel-which-requires-authentication-api-called-first)  in karavan? 

As far as I understand it has 2 methods one to set the token which is triggered from a processor and the other is get the token

When I tried to do the same in  karavan, I needed to mention the class name which results into class not found error

**Comment by davsclaus** [comment]
There is the swift dataformat that has some support for finance protocols
https://camel.apache.org/components/4.0.x/dataformats/index.html
